src/BuildingRegistry/Building/Events/BuildingOutlineWasChanged.cs

```diff
@@ -0,0 +1,78 @@
+namespace BuildingRegistry.Building.Events
+{
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using Be.Vlaanderen.Basisregisters.EventHandling;
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using Newtonsoft.Json;
+
+    [EventTags(EventTag.For.Sync, EventTag.For.Edit, Tag.Building)]
+    [EventName(EventName)]
+    [EventDescription("Het gebouw werd ingeschetst (via wijziging).")]
+    public class BuildingOutlineWasChanged : IBuildingEvent
+    {
+        public const string EventName = "BuildingOutlineWasChanged"; // BE CAREFUL CHANGING THIS!!
+
+        [EventPropertyDescription("Objectidentificator van het gebouw.")]
+        public int BuildingPersistentLocalId { get; }
+
+        [EventPropertyDescription("Objectidentificatoren van de gekoppelde gebouweenheden met positieGeometrieMethode afgeleidVanObject.")]
+        public IEnumerable<int> BuildingUnitPersistentLocalIds { get; }
+
+        [EventPropertyDescription("Extended WKB-voorstelling van de gebouwgeometrie (Hexadecimale notatie).")]
+        public string ExtendedWkbGeometryBuilding { get; }
+
+        [EventPropertyDescription("Extended WKB-voorstelling van de gebouweenheidpositie van de gekoppelde gebouweenheden met positieGeometrieMethode afgeleidVanObject (Hexadecimale notatie).")]
+        public string? ExtendedWkbGeometryBuildingUnits { get; }
+
+        [EventPropertyDescription("Metadata bij het event.")]
+        public ProvenanceData Provenance { get; private set; }
+
+        public BuildingOutlineWasChanged(
+            BuildingPersistentLocalId buildingPersistentLocalId,
+            IEnumerable<BuildingUnitPersistentLocalId> buildingUnitPersistentLocalIds,
+            ExtendedWkbGeometry extendedWkbGeometryBuilding,
+            ExtendedWkbGeometry? extendedWkbGeometryBuildingUnits)
+        {
+            BuildingPersistentLocalId = buildingPersistentLocalId;
```

Comment:
So if I'm reading the PR correctly. If `BuildingUnitPersistentLocalIds` is empty, then `ExtendedWkbGeometryBuildingUnits` will be null and vice versa?
Can we add a check in here? If one of the two is null/empty and the other isn't, `throw new ArgumentNullException` ?

Reply:
That's correct.

Reply:
Added check.